components/SpotifyInterface/SpotifyAPICall.c

```diff
@@ -2,21 +2,16 @@
 #include "SpotifyTypedef.h"
 
 static const char *TAG = "HTTP";
-QueueHandle_t httpToSpotifyDataQueue = NULL;
+SpotifyAPIBuffer_t SpotifyAPIBuffer;
 
-// Include the server root certificate data
-extern const uint8_t server_root_cert_pem_start[] asm("_binary_server_root_cert_pem_start");
-extern const uint8_t server_root_cert_pem_end[] asm("_binary_server_root_cert_pem_end");
-
-// Include the local server certificate and private key data
-extern const uint8_t local_server_cert_pem_start[] asm("_binary_local_server_cert_pem_start");
-extern const uint8_t local_server_cert_pem_end[] asm("_binary_local_server_cert_pem_end");
-extern const uint8_t local_server_key_pem_start[] asm("_binary_local_server_key_pem_start");
-extern const uint8_t local_server_key_pem_end[] asm("_binary_local_server_key_pem_end");
+void SpotifyAPICallInit(char *messageBuffer,SemaphoreHandle_t spotifyResponseReady)
+{
+    SpotifyAPIBuffer.MessageBuffer = messageBuffer;
+    SpotifyAPIBuffer.SpotifyResponseReadyFlag = spotifyResponseReady;
+}
 
 esp_err_t HttpEventHandler(esp_http_client_event_t *evt) 
 {
```

Comment:
how mange receive incomplete data, can you detect it ? if you receive incomplete data, throw it away